### PR TITLE
Feature/io 493 splash screen

### PR DESCRIPTION
**Pull request by ekaterina-demeshko-idf**

### CODE REVIEW

File: packages/presentation/lib/base/bloc.dart [hunk 1]
```diff
@@ -38,7 +38,5 @@ abstract class BlocImpl<T extends BaseArguments, D> implements Bloc<T, D> {
   void initArgs(T arguments) {}
 
   @override
-  void dispose() {
-    _data.close();
-  }
+  void dispose() {}
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/base/bloc.dart]: ?

**Review by nikolay-ahramovich-idf — APPROVED** [review]

File: packages/presentation/lib/base/bloc.dart (hunk 1 above)
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/base/bloc.dart]: why did u remove closing the stream?

File: packages/presentation/lib/my_app.dart [hunk 2]
```diff
@@ -1,18 +1,45 @@
...
       debugShowCheckedModeBanner: false,
       theme: ThemeData(
         primarySwatch: Colors.deepPurple,
+        canvasColor: const Color(PrimaryColors.primaryBackgroundColor),
+      ),
+      home: StreamBuilder<BlocData>(
+        stream: bloc.dataStream,
+        builder: (context, result) {
+          final blocData = result.data;
+          final appData = blocData?.data;
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/my_app.dart]: no need in blocData variable. use final appData = blocData?.data?.data;

File: packages/presentation/lib/screen/splash/splash_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  Future<void> loadData() async {
+    Future.delayed(
+      const Duration(seconds: 3),
+      () => appNavigator.popAndPush(
+        HomeScreen.page(
+          HomeScreenArguments(),
+        ),
+      ),
+    );
+  }
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/splash/splash_bloc.dart]: move the delay logic into usecase. in this case u will have

await immitateApiCallUseCase();
navigator.popAndPush();

File: packages/presentation/lib/screen/splash/splash_data.dart [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class SplashData {
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/splash/splash_data.dart]: no need in it

File: packages/presentation/lib/screen/splash/splash_screen.dart [hunk 5]
```diff
@@ -13,30 +16,36 @@ class SplashScreen extends StatelessWidget {
...
-            ),
-          ),
-          child: const Center(
-            child: Icon(
-              Icons.energy_savings_leaf_outlined,
-              color: Colors.white,
-              size: 162,
+    return StreamBuilder(
+      stream: bloc.dataStream,
+      builder: (context, snapshot) {
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/splash/splash_screen.dart]: no need in streambuilder

File: packages/presentation/lib/utils/widgets/tabbar.dart [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+              child: TabBarHomeScreen(tabController: tabController),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class TabBarHomeScreen extends StatelessWidget {
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/utils/widgets/tabbar.dart]: move into separate file

File: pubspec.yaml [hunk 7]
```diff
@@ -15,6 +15,7 @@ dependencies:
     path: packages/presentation
   data:
     path: packages/data
+  flutter_svg: ^1.1.4
```
Comment (sergei-mikhailovskii-idf) [inline pubspec.yaml]: do u need it here?

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/presentation/lib/base/bloc.dart (hunk 1 above)
Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/base/bloc.dart]: it conflicted with `_data.add(_blocData.copy<D>());` handleData method

Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/base/bloc.dart]: [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: Bad state: Cannot add event after closing

File: packages/presentation/lib/screen/home/home_screen.dart [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+              appBar: AppBar(
+                  backgroundColor:
+                      const Color(PrimaryColors.primaryBackgroundColor),
+                  title: const Text(
+                    'Star Movie',
+                    style: TextStyle(
+                        fontSize: 28,
+                        fontWeight: FontWeight.bold,
+                        color: Colors.white,
+                        fontFamily: '.SF Pro Text'),
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screen/home/home_screen.dart]: let's move it several UI utils classes, I'll send u an example

File: packages/presentation/lib/utils/widgets/tabbar.dart [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    required this.tabController,
+  }) : super(key: key);
+
+  final TabController tabController;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      // height: 50,
+      width: MediaQuery.of(context).size.height,
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/utils/widgets/tabbar.dart]: maybe better double.infinity?

File: pubspec.yaml (hunk 7 above)
Comment (andrei-lastochkin-idf) [reply pubspec.yaml]: no, u don't need it here

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: packages/presentation/lib/base/bloc.dart (hunk 1 above)
Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/base/bloc.dart]: it conflicted with `_data.add(_blocData.copy<D>()); `handleData method


`[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: Bad state: Cannot add event after closing`

File: pubspec.yaml (hunk 7 above)
Comment (ekaterina-demeshko-idf) [reply pubspec.yaml]: removed

File: packages/presentation/lib/base/bloc.dart (hunk 1 above)
Comment (sergei-mikhailovskii-idf) [reply packages/presentation/lib/base/bloc.dart]: need to resolve the error. seems that u try to push event after dispose

Comment (sergei-mikhailovskii-idf) [reply packages/presentation/lib/base/bloc.dart]: need to resolve the error. seems that u try to push event after dispose

File: packages/presentation/lib/screen/splash/splash_screen.dart [hunk 10]
```diff
@@ -23,29 +23,24 @@ class SplashScreen extends StatefulWidget {
...
+              colors: [
+                Color(PrimaryColors.primaryColor),
+                Color(PrimaryColors.primaryGradientBottom),
+              ],
             ),
           ),
-        );
-      },
+          child: Center(
+            child: SvgPicture.asset('resources/images/Group.svg'),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/splash/splash_screen.dart]: move images paths to separate class

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/presentation/lib/base/bloc.dart (hunk 1 above)
Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/base/bloc.dart]:
```
E/flutter (12836): [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: Bad state: Cannot add event after closing
E/flutter (12836): #0      _StreamController.add (dart:async/stream_controller.dart:595:24)
E/flutter (12836): #1      BlocImpl.handleData (package:presentation/base/bloc.dart:31:11)
E/flutter (12836): #2      _AppBloc._updateData (package:presentation/app/app_bloc.dart:102:11)
E/flutter (12836): #3      _AppBloc._push (package:presentation/app/app_bloc.dart:46:5)
E/flutter (12836): #4      _AppBloc._popAndPush (package:presentation/app/app_bloc.dart:66:5)
E/flutter (12836): #5      _AppNavigatorImpl.popAndPush (package:presentation/navigation/app_navigator.dart:100:50)
E/flutter (12836): #6      _SplashBloc.loadData (package:presentation/screen/splash/splash_bloc.dart:26:18)
E/flutter (12836): <asynchronous suspension>
```

Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/base/bloc.dart]: fixed

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**